### PR TITLE
Add instructions to install GSL

### LINKED ISSUE #263: Installation instructions missing GSL requirement
For installing sgkit, we need GSL (GNU Scientific Library) to be installed, which is a requirement for `msprime`:
https://msprime.readthedocs.io/en/stable/installation.html#requirements and `msprime` is a requirement for sgkit: https://github.com/pystatgen/sgkit/blob/c18b068a46838070757ac3b83b7f4c1ecc29dec5/requirements-dev.txt#L9

https://github.com/pystatgen/sgkit/blob/c18b068a46838070757ac3b83b7f4c1ecc29dec5/.github/workflows/build.yml#L20-L22

This is missing in the contributing docs:
https://pystatgen.github.io/sgkit/contributing.html#development-environment

### PR DESCRIPTION
**Pull request by aktech**
Fixes #263

cc @tomwhite @ravwojdyla

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -94,6 +94,12 @@ pip::
   cd sgkit
   pip install -r requirements.txt -r requirements-dev.txt -r requirements-doc.txt
 
+You also need to install GSL (GNU Scientific Library), which is required for ``msprime``.
+Since its a system package and installation may vary depending on the operating
+system, the easiest way to install it is using conda::
+
+    conda install -c conda-forge gsl
```
Comment (ravwojdyla) [inline docs/contributing.rst:101]: It feels a bit weird that we first recommend using pip above and then after it likely fails on `msprime` install, we bring in `conda`. I wonder if we should instead link to [msprime doc](https://msprime.readthedocs.io/en/stable/installation.html#installing-system-requirements) which has extensive information about GSL install options?

Comment (aktech) [reply docs/contributing.rst:101]: True, that is weird indeed. I'll add a link to msprime doc rather.

**Review by ravwojdyla — CHANGES_REQUESTED** [review]
during our weekly sync, we have discussed that this should be moved up a bit.

**Review by ravwojdyla — APPROVED** [review]

**Comment by aktech** [comment]
Thanks for the review @ravwojdyla, can we merge this?

**Comment by ravwojdyla** [comment]
@aktech np. let me summon the @pystatgen/committers :)

**Pull request merged**